### PR TITLE
Adds a short practical summary of ByzCoin

### PR DESCRIPTION
**Pull request by nkcr**
**What this PR does**

*See commit message*

### CODE REVIEW

File: byzcoin/README.md [hunk 1]
```diff
@@ -14,6 +14,43 @@ them as seem fit.
...
 use them. It should grow over time as more parts of the system are implemented.
 
+## Short summary of ByzCoin
+
+- ByzCoin is a **distributed ledger**: a database that is
+  distributed across several nodes, and where the authority to add changes
+  to the database is decentralized.
+- To ensure that updates to the database are strictly ordered, ByzCoin
+  uses a **blockchain** called a **Skipchain**: every node has its local copy of the ledger
+  (database). Every change to the ledger is packed into a block that is
```
Comment (ineiti) [inline byzcoin/README.md]: The novelties of skipchains are the following:
- delegation of trust from one block to the next: as the roster is stored in the block, the current nodes delegate trust to the next set of nodes
- finality using forwardlinks: where other blockchains use proof-of-work and longest chain to define the latest state of the system, skipchains never change and don't allow for split views (except twice in the 9cc3 skipchain, due to an error ;)
- short proofs: using only the hash of the genesis block (9cc3...), every node can return a proof that the latest block is valid by sending only the forward-links. In other systems you need to have your own trusted node follow the skipchain so you can be sure that the latest block is valid

and one thing that is not implemented, but describe in `Chainiac`:
- using multiple skipchains, where one holds the definition of the roster, while the other skipchain holds the actual transactions.

File: byzcoin/README.md [hunk 2]
```diff
@@ -14,6 +14,43 @@ them as seem fit.
...
+  cryptographically chained to the previous one, and forward links in
+  the chain represent consensus among the verifiers that these changes
+  belong in the database.
+- In ByzCoin, the data are organised as instances of **Smart
```
Comment (ineiti) [inline byzcoin/README.md]:
```suggestion
- In ByzCoin, the data is organised as instances of **Smart
```

File: byzcoin/README.md [hunk 3]
```diff
@@ -14,6 +14,43 @@ them as seem fit.
...
+  Contracts**, where a smart contract can be seen as a class and an
+  instance an object of this class. In ByzCoin, one can *Spawn* a new
```
Comment (ineiti) [inline byzcoin/README.md]:
```suggestion
  instance as an object of this class. In ByzCoin, one can *Spawn* a new
```

File: byzcoin/README.md [hunk 4]
```diff
@@ -14,6 +14,43 @@ them as seem fit.
...
+  instance of a contract, *Invoke* a command (or method) on an existing
+  instance, or *Delete* an instance. *Spawn*, *Invoke*, and *Delete*
+  represent all the possible actions a user can do to update the ledger.
+- Every request to update the ledger is a transaction made up of one
+  or more instructions. The transactionn is sent to one of the nodes.
```
Comment (ineiti) [inline byzcoin/README.md]:
```suggestion
  or more instructions. The transaction is sent to one of the nodes.
```

File: byzcoin/README.md [hunk 5]
```diff
@@ -14,6 +14,43 @@ them as seem fit.
...
+  All instructions in the transaction must be approved by a quorum of
+  the nodes, or else the entire transaction is refused.
+- As an example, we can have a simple *Project* smart contract that has
```
Comment (ineiti) [inline byzcoin/README.md]: I would not put the example in the `Short summary`, but rather in it's own `## Example`

File: byzcoin/README.md [hunk 6]
```diff
@@ -14,6 +14,43 @@ them as seem fit.
...
+  the status's change is eventually written in a block, allowing anyone to
+  track and witness the evolution of the `state` of the *Project* in the ledger.
+  If the smart contract enforces a rule that `status: pending`
+  must always be followed by `status: active`, then the propsed transaction
+  will be refused by nodes which are honestly implementing the smart contract.
+  Even if a minority of the nodes are dishonestly allowing the transition
+  directly to `state: done`, they will not form a quorum, and the incorrect
+  state change cannot be introduced into the database.
+
 ## Overview
```
Comment (ineiti) [inline byzcoin/README.md]: And we should update the overview. I know of at least one error in the figure, and some things might be outdated a bit.

File: byzcoin/README.md (hunk 1 above)
Comment (nkcr) [reply byzcoin/README.md]: that's great, I'm adding it to the `skipchain/README`

File: byzcoin/README.md (hunk 2 above)
Comment (nkcr) [reply byzcoin/README.md]: that's a debate, but it seems there is nothing wrong with this plural form of "data".

File: byzcoin/README.md (hunk 5 above)
Comment (nkcr) [reply byzcoin/README.md]: We should then provide a more complete example. To keep it simple I prefer if we keep it there.

File: byzcoin/README.md (hunk 6 above)
Comment (nkcr) [reply byzcoin/README.md]: that would be awesome! Do you want to suggest an update of the overview that we include in this PR?

File: byzcoin/README.md (hunk 2 above)
Comment (ineiti) [reply byzcoin/README.md]: I was not aware that there is a discussion around this - https://www.theguardian.com/news/datablog/2010/jul/16/data-plural-singular

According to that article you're more of a traditionalist, though ;)

**Comment by ineiti** [comment]
Here you go - only corrected the things I could easily find. Also included @cgrigis latest changes to `Instruction.Delete` which includes `Arguments` now.

[ByzCoin.svg.zip](https://github.com/dedis/cothority/files/4244305/ByzCoin.svg.zip)

**Comment by ineiti** [comment]
Here you go (this replaces the one from my previous comment):

[ByzCoin-2.svg.zip](https://github.com/dedis/cothority/files/4244326/ByzCoin-2.svg.zip)

**Comment by ineiti** [comment]
I wonder whether we should also add a paragraph about DARCs for the access control?

**Pull request merged**